cincinnati/src/lib.rs

```diff
@@ -940,49 +940,63 @@ pub mod testing {
     /// Compares two Graphs and gives a verbose error if not.
     pub fn compare_graphs_verbose(
         mut left: Graph,
+        left_name: &str,
         mut right: Graph,
+        right_name: &str,
         unwanted_metadata_keys: &[&str],
     ) -> Result<(), Box<dyn std::error::Error>> {
         let removed_keys_left = remove_release_metadata(&mut left, unwanted_metadata_keys);
         let removed_keys_right = remove_release_metadata(&mut right, unwanted_metadata_keys);
+        let edges_left = right.get_edges(false)?;
+        let edges_right = left.get_edges(false)?;
+        let releases_left: std::collections::BTreeSet<Release> = (&left).into();
+        let releases_right: std::collections::BTreeSet<Release> = (&right).into();
 
-        if left == right {
+        if edges_left == edges_right && releases_left == releases_right {
             return Ok(());
         }
 
         let mut output: Vec<String> = vec![];
-        output.push("Graphs differ! Showing differences from left to right".into());
-        output.push("-----------------------------------------------------".into());
-
-        let edges_left = right.get_edges(false)?;
-        let edges_right = left.get_edges(false)?;
-        output.push("edges: ".into());
         output.push(
-            prettydiff::diff_lines(
-                &serde_json::to_string_pretty(&edges_left)?,
-                &serde_json::to_string_pretty(&edges_right)?,
+            format!(
+                "Graphs differ! Showing differences from {} to {}",
+                left_name, right_name
             )
-            .format(),
+            .into(),
         );
 
-        let releases_left: std::collections::BTreeSet<Release> = (&left).into();
-        let releases_right: std::collections::BTreeSet<Release> = (&right).into();
-        output.push("nodes:".into());
-        output.push(
-            prettydiff::diff_lines(
-                Box::leak(Box::new(serde_json::to_string_pretty(&releases_left)?)),
-                Box::leak(Box::new(serde_json::to_string_pretty(&releases_right)?)),
-            )
-            .format(),
-        );
+        if edges_left != edges_right {
```

Comment:
there are a few changes made by `cincinnati/src/lib: Only print divergent portions of the diff` (currently 6c9accbe).  Instead of running separate threads on all the lines it touches, we can use this thread to debate whether we want to keep the commit or drop it.  I added it because the ANSI color coding out of `prettydiff::diff_lines` is not particularly convenient for me to consume, and between the effort of dumping the build log into a terminal to get the colors rendered and the false-positives around graph differences before 419246b03f (cincinnati/src/lib: Ignore node order in graph comparison, [thread][1]), it was easier for me to find the remaining bits of the fixtures I needed to touch if I did not print out the matching sections.  I am agnostic about whether we land this commit in the final PR, but it is helping me with the fixture iterations, so even if the consensus is "drop it before we merge the PR", I will keep it around until the fixtures are green.

[1]: https://github.com/openshift/cincinnati/pull/314#discussion_r515282649

Reply:
I've [dropped][1] this change.

[1]: https://github.com/openshift/cincinnati/pull/314#issuecomment-720002421